__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -151,6 +248,172 @@ describe('start', () => {
+    hWallet = await generateWalletHelper({
+      seed: walletData.words,
+      preCalculatedAddresses: walletData.addresses,
+      tokenUid,
+    });
+    expect(hWallet.isReady()).toStrictEqual(true); // This operation should work
+
+    // Now testing the methods that use this set tokenUid information
+    // FIXME: No need to explicitly pass the non-boolean `false` as a tokenUid to get this result.
+    await expect(await hWallet.getBalance(false)).toStrictEqual([
+      {
+        token: {
+          id: tokenUid,
+          name: "",
+          symbol: ""
+        },
+        balance: {
+          unlocked: 100,
+          locked: 0
+        },
+        transactions: 3,
+        lockExpires: null,
+        tokenAuthorities: {
+          unlocked: {
+            mint: true,
+            melt: true
+          },
+          locked: {
+            mint: false,
+            melt: false
+          }
+        }
+      }
+    ]);
+
+    // FIXME: We should not have to explicitly pass an empty token uid to get this result
+    const txHistory1 = await hWallet.getTxHistory({ token_id: undefined });
+    expect(txHistory1).toStrictEqual([
+      expect.objectContaining({
+        txId: tokenUid,
+        tokenUid: tokenUid,
+        balance: 100
+      }),
+    ]);
+
+    /*
+     * These tests could be created inside the `getBalance` and `getTxHistory` sections but for
+     * simplicity sake, since they are so small, were added here just as a complement to
+     * this `start` test.
+     */
+  });
+
+  it('should start a wallet via xpub', async () => {
+    const walletData = precalculationHelpers.test.getPrecalculatedWallet();
+    const xpriv = walletUtils.getXPrivKeyFromSeed(
+      walletData.words,
+      { networkName: NETWORK_NAME })
+    const privkey = walletUtils.deriveXpriv(xpriv, "0'/0");
+    const xpub = privkey.xpubkey;
+
+    // Creating a new wallet with a known set of words just to generate the custom token
+    let hWallet = await generateWalletHelper({ xpub });
+    expect(hWallet.isReady()).toStrictEqual(true);
+
+    // Validating that methods that require the private key will throw on call
+    await expect(hWallet.consolidateUtxos()).rejects.toThrow(WalletFromXPubGuard);
+    await expect(hWallet.sendTransaction()).rejects.toThrow(WalletFromXPubGuard);
+    await expect(hWallet.sendManyOutputsTransaction()).rejects.toThrow(WalletFromXPubGuard);
+    await expect(hWallet.prepareCreateNewToken()).rejects.toThrow(WalletFromXPubGuard);
+    await expect(hWallet.prepareMintTokensData()).rejects.toThrow(WalletFromXPubGuard);
+    await expect(hWallet.prepareMeltTokensData()).rejects.toThrow(WalletFromXPubGuard);
+    await expect(hWallet.prepareDelegateAuthorityData()).rejects.toThrow(WalletFromXPubGuard);
+    await expect(hWallet.prepareDestroyAuthorityData()).rejects.toThrow(WalletFromXPubGuard);
+
+    // Validating that the address generation works as intended
```

Comment:
I would validate that you can get the wallet balance and utxos correctly

Reply:
Added these tests in [cc7fb5d](https://github.com/HathorNetwork/hathor-wallet-lib/pull/409/commits/cc7fb5d017b78f0caad96a470122eb0b531b5458).